### ISSUE TITLE
Async event listener not invoked

### ISSUE THREAD

**Issue by guansss** [sampotts/plyr#2199]
### Expected behaviour
Using an async function as event listener.

### Actual behaviour
The async function will never be invoked.

### Steps to reproduce

```js
player.on('ready', async () => {
  console.log('ready');
});
```

I looked into the code and realized the reason is that the async function didn't pass the `isFunction()` check. It checks if the given input's constructor is `Function`, then fails because `(async () => {}).constructor === Function` is false.

IMO it's better to use `typeof input === 'function'`, which works for async functions.

### Environment

- Browser: Chrome
- Version: 90.0.4430.93

**Comment by dy** [comment]
Yep. It is useful to unwrap events to async flows, but now it is blocked:
```

await event(player, 'ready')

// never reached
player.currentTime = 0.1

await event(player, 'timeupdate')

player.elements.container.classList.add('plyr--ready')


// wait until event
function event (target, evt) {return new Promise(r => target.on(evt, function fn(){
  target.off(evt, fn),r()
}))}
```
